### PR TITLE
Make map parser

### PR DESCRIPTION
**Pull request by blauks**
Started on making a map parser, to create paths from files. 

Not quite finished, and I want to check if anyone has any feature requests. I for example wondered if I should implement a function that loops through the files in the maps folder, to find all the .map files and puts them into the game so one can choose which map they want to play on. This will probably make the modifiability of the project a bit higher maybe?!

### CODE REVIEW

**Comment by fredrikbw** [comment]
In order for the towers to find the unit that it next should attack, it would be nice to have a function that gives an Array of x and y of each tile that is on the path, ordered from start to finish.

**Comment by blauks** [comment]
> In order for the towers to find the unit that it next should attack, it would be nice to have a function that gives an Array of x and y of each tile that is on the path, ordered from start to finish.

I'm on it boss. I will even try to use weird functional coding things instead of for loops to make andreas happy 👍

File: client/core/src/com/cows/game/Map.kt [hunk 1]
```diff
@@ -2,47 +2,74 @@ package com.cows.game
...
+                0000000100000111
+                0000000100000000
+                1111111100000000
+                0000000000000000
+            """.trimMargin())
+        }
+        return generatePath(map)
+    }
+
+    private fun rawMapToArray(rawMap: String): Array<IntArray> {
```
Comment (AndreasWintherMoen) [inline client/core/src/com/cows/game/Map.kt:45]: The whole rawMapToArray function can become a one liner (if we want to). _Array(HEIGHT) { x -> IntArray(WIDTH) { y -> rawMap[x * WIDTH + y].digitToInt() } }_

**Comment by AndreasWintherMoen** [comment]
I think it looks good! We can simplify rawMapToArray a bit, but this solution works as well.

File: client/core/src/com/cows/game/Map.kt (hunk 1 above)
Comment (blauks) [reply client/core/src/com/cows/game/Map.kt:45]: We don't want to:/

**Comment by blauks** [comment]
@fredrikbw Something like this?

**Comment by AndreasWintherMoen** [comment]
> @fredrikbw Something like this?

Yesh! I assume it works? :D

File: client/core/src/com/cows/game/Map.kt (hunk 1 above)
Comment (fredrikbw) [reply client/core/src/com/cows/game/Map.kt:45]: > @fredrikbw Something like this?

Yeah! Thats great 💯

**Comment by fredrikbw** [comment]
@blauks It works! 💯

**Review by AndreasWintherMoen — APPROVED** [review]
Looking good! Will be nice to get this feature into master.

**Pull request merged**